metrics/serve_test.go

```diff
@@ -414,3 +414,49 @@ func TestRunMetricsSpikeChange(t *testing.T) {
 		}
 	}
 }
+
+func TestCollectorLabels(t *testing.T) {
+	testCfg := Config{
+		GaugeMetricCount:    1,
+		LabelCount:          2,
+		SeriesCount:         100,
+		MaxSeriesCount:      30,
+		MinSeriesCount:      10,
+		SpikeMultiplier:     1.5,
+		MetricLength:        1,
+		LabelLength:         1,
+		SeriesOperationMode: spikeOpMode,
+		ConstLabels:         []string{"constLabel=test"},
+	}
+
+	assert.NoError(t, testCfg.Validate())
+
+	reg := prometheus.NewRegistry()
+	col := NewCollector(testCfg)
+	reg.MustRegister(col)
+
+	go col.Run()
+	t.Cleanup(func() {
+		col.Stop(nil)
+	})
+
+	time.Sleep((2 * time.Second))
```

Comment:
Do you mind using some kind of wait e.g. `assert.Eventually` instead of sleep?

Reply:
Thanks for the feedback @bwplotka it makes sense , I have updated the test to use the `c.upateNotifyCh` for assertions